sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_pyamqp_transport.py

```diff
@@ -0,0 +1,561 @@
+            retry_mode=config.retry_mode,  # pylint:disable=protected-access
+            no_retry_condition=NO_RETRY_ERRORS,
+            custom_condition_backoff=CUSTOM_CONDITION_BACKOFF,
+        )
+
+    @staticmethod
+    def create_link_properties(link_properties):
+        """
+        Creates and returns the link properties.
+        :param dict[bytes, int] link_properties: The dict of symbols and corresponding values.
+        :rtype: dict
+        """
+        return {symbol: utils.amqp_long_value(value) for (symbol, value) in link_properties.items()}
+
+    @staticmethod
+    def create_connection(**kwargs):
+        """
+        Creates and returns the uamqp Connection object.
+        :keyword str host: The hostname, used by uamqp.
+        :keyword JWTTokenAuth auth: The auth, used by uamqp.
+        :keyword str endpoint: The endpoint, used by pyamqp.
+        :keyword str container_id: Required.
+        :keyword int max_frame_size: Required.
+        :keyword int channel_max: Required.
+        :keyword int idle_timeout: Required.
+        :keyword Dict properties: Required.
+        :keyword int remote_idle_timeout_empty_frame_send_ratio: Required.
+        :keyword error_policy: Required.
+        :keyword bool debug: Required.
+        :keyword str encoding: Required.
+        """
+        endpoint = kwargs.pop("endpoint")
+        host = kwargs.pop("host")  # pylint:disable=unused-variable
+        auth = kwargs.pop("auth")  # pylint:disable=unused-variable
+        network_trace = kwargs.pop("debug")
+        return Connection(endpoint, network_trace=network_trace, **kwargs)
+
+    @staticmethod
+    def close_connection(connection):
+        """
+        Closes existing connection.
+        :param connection: uamqp or pyamqp Connection.
+        """
+        connection.close()
+
+    @staticmethod
+    def get_connection_state(connection):
+        """
+        Gets connection state.
+        :param connection: uamqp or pyamqp Connection.
+        """
+        return connection.state
+
+    @staticmethod
+    def create_send_client(*, config, **kwargs):  # pylint:disable=unused-argument
+        """
+        Creates and returns the uamqp SendClient.
+        :param ~azure.eventhub._configuration.Configuration config: The configuration.
+
+        :keyword str target: Required. The target.
+        :keyword JWTTokenAuth auth: Required.
+        :keyword int idle_timeout: Required.
+        :keyword network_trace: Required.
+        :keyword retry_policy: Required.
+        :keyword keep_alive_interval: Required.
+        :keyword str client_name: Required.
+        :keyword dict link_properties: Required.
+        :keyword properties: Required.
+        """
+
+        target = kwargs.pop("target")
+        # TODO: extra passed in to pyamqp, but not used. should be used?
+        msg_timeout = kwargs.pop("msg_timeout")  # pylint: disable=unused-variable  # TODO: not used by pyamqp?
```

Comment:
Is the plan to investigate this in this PR or separate? (I'd probably vote separate...)

Reply:
yep! separate PR and issue created for this here: https://github.com/Azure/azure-sdk-for-python/issues/25875